_posts/2023-03-12-gs_00008_programming_II.md

```diff
@@ -0,0 +1,188 @@
+Parallel programs are similar, but here we want to assume pure sequential
+programs.
+
+Let's start with the eggs. We have to take an egg, break it, and put it into a
+bowl. Also, we need to throw the eggshells into the trash. For the second egg,
+we have to do exactly the same. In a programming language, you can define a
+**function** _put_egg_in_bowl_ to define a sequence that is used multiple times.
+Next, you can **call** this function in a **loop** for every egg. So, a program
+to prepare the eggs could look like this:
+
+    function put_egg_in_bowl:
+        take egg
+        break egg
+        put it into a bowl
+        throw shell into trash
+
+    for 2 times:
+        put_egg_in_bowl
+
+Note, that we use indentation (here we use 4 spaces) to show which instructions
+belong to our function and loop. Also, note that this is not a real programming
+language, but it serves our demonstration purpose.
+
+After the eggs, we want to weigh the flour. How do you usually do it? You watch
+the scale. While your target weight of 200g is not reached, you put more flour
+into the bowl. In programming languages, there exist two types of loops. For the
+one we used above, we know how often we need to repeat it (two times in our
+example). This is called a **for-loop**. For weighing the flour, this is not
+very useful since you don't know how often you have to put more. Instead, we
+need a loop that continues while a _condition_ is true. This is called
+**while-loop**:
+
+    while flour < 200g:
+        put more flour
+
+You can read it like this: "Repeat putting more flour while the weight is less
+than 200g." Once the weight of the flour reaches 200g, the while-loop is done.
+We can continue very similarly for the milk:
+
+    while milk < 200ml:
+        put more milk
+
+_flour_ and _milk_ represent the current amount. In programming languages, we
+call these **variables**. In simple words, variables can store any kind of
+information.
+
+<div class='note-exercise' markdown="1">
+Exercise:
+
+Can you continue with the rest of the pancake?
+
+</div>
+
+While you're finally eating the pancake you often ask another question: "Do you
+like the pancake?" If yes, you might want to recommend it to a friend. To
+implement such a question you can use an **if-else-statement**:
+
+    if pancake is delicious:
+        recommend the recipe
+    else:
+        delete the recipe
+
+The if statement checks whether a condition is true. In our example, it is
+checked whether the pancake is delicious. If that condition is true, the
+instructions indented below the if statement will be executed, in our case
+recommending the recipe. If the pancakes don't taste good, the instructions
+below the else statement will be executed, here, deleting the recipe.
+
+You might say that this is too easy and this can't be all. Well... Yes and No!
+Of course, there is much more to learn. Otherwise, software engineers wouldn't
```

Comment:
Don't know if that "paid well" sentence is suitable. I actually think it would be ok to leave it out. You could write something like:

> "Of course, there is much more to learn. But it turns out that even the most complex programs all still make use of those basic principles we've presented you above!"